### PR TITLE
Remove oldmodel macro use

### PR DESCRIPTION
**Pull request by zuckschwerdt**
This removes the deprecated "oldmodel" workaround macro `_X()`.

Regression checks pass, but if anyone can spare some time then please review.

### CODE REVIEW

File: src/devices/acurite.c [hunk 1]
```diff
@@ -1364,24 +1363,18 @@ r_device acurite_th = {
  */
 static char *acurite_txr_output_fields[] = {
         "model",
-        "subtype",      // TODO: remove this
         "message_type", // TODO: remove this
```
Comment (Mindavi) [inline src/devices/acurite.c:1366]: This one also has 'TODO: remove this'

File: src/devices/bt_rain.c [hunk 2]
```diff
@@ -75,7 +75,7 @@ static int bt_rain_decode(r_device *decoder, bitbuffer_t *bitbuffer)
             "battery",          "Battery",          DATA_STRING, battery ? "LOW" : "OK",
             "transmit",         "Transmit",         DATA_STRING, button ? "MANUAL" : "AUTO", // TODO: delete this
```
Comment (Mindavi) [inline src/devices/bt_rain.c:76]: Has a TODO: delete this, replaced by the 'button' value, I guess.

File: src/devices/bt_rain.c [hunk 3]
```diff
@@ -91,7 +91,6 @@ static char *output_fields[] = {
         "battery",
         "transmit", // TODO: delete this
```
Comment (Mindavi) [inline src/devices/bt_rain.c:92]: Can be removed as well if it's removed above.

**Review by Mindavi — APPROVED** [review]
Thanks for doing this cleanup. I put some comments on some missed TODOs, I think those would be good to check before merging. Looks good in general.

I did look at most of this, but not everything. I don't see anything surprising though.

**Comment by obones** [comment]
Shouldn't this be the occasion to also make sure all the "battery" string fields are replaced by "battery_ok" int fields?
Or should it be done in another pull request?

**Comment by zuckschwerdt** [comment]
Thanks. Yes, since there is no oldmodel switch anymore there's no reason to have old battery fields. I'll make a second PR for a clean history without merge commits.
I'll look into the remaining "remove this" notes afterwards.

**Pull request merged**